src/main.cpp

```diff
@@ -2148,6 +2170,16 @@ bool CheckTxInputs(const CTransactionBase& txBase, CValidationState& state, cons
                 REJECT_INVALID, "bad-txns-inputvalues-outofrange");
     }
 
+    try {
+        nValueIn += txBase.GetCSWValueIn();
+        if (!MoneyRange(nValueIn))
+            return state.DoS(100, error("CheckInputs(): Total inputs value out of range."),
+                REJECT_INVALID, "bad-txns-inputvalues-outofrange");
+    } catch (const std::runtime_error& e) {
```

Comment:
I think you do not need the try/catch here currently. There is no cast or anything. Remove it